lib/ruby_lsp/tapioca/addon.rb

```diff
@@ -71,6 +74,21 @@ def workspace_did_change_watched_files(changes)
           path = URI(change[:uri]).to_standardized_path
           next if path.end_with?("_test.rb", "_spec.rb")
 
+          case change[:type]
+          when Constant::FileChangeType::CREATED, Constant::FileChangeType::CHANGED
+            content = File.read(path)
+            current_checksum = Zlib.crc32(content).to_s
+
+            if change[:type] == Constant::FileChangeType::CHANGED && @file_checksums[path] == current_checksum
+              $stderr.puts "File has not changed. Skipping #{path}"
```

Comment:
This _might_ be a little noisy to show every time, but we can keep it for now.

Reply:
Good point. Let's monitor it and I'll remove if it's too much.